10_rest/nickolee/README.md

```diff
@@ -0,0 +1,103 @@
+# Project: Final Step in Incrementally Building Towards a Puppy REST API
+
+This project is part of the [go course](https://github.com/anz-bank/go-course/) run by ANZ Bank.
+
+It builds you up incrementally with the ultimate end goal of having a fully functioning Rest API.
+
+And finally after a deep and delightful learning journey we have finally arrived at Lab 10!
+
+It aims to impart several key learnings such as but not limited to:
+
+- Working with custom types in go using structs
+- Understanding and working with interfaces in go
+- Implementing multiple implementations of an interface
+- Working with various tools in the Go toolchain such as `golangci-lint` and `go test -coverprofile=coverage.out` to check for 100% test coverage or fail CI
+- Go standard project layout
+- Best practices with Git. Proper commit messages and PRs. Git rebasing when necessary to keep commit history clean etc.
+- Creating custom error types to suit your project needs
+- Thread safety using sync.Mutex to prevent errors associated with data race conditions
+- Test coverage using [test suites] (https://godoc.org/github.com/stretchr/testify/suite) as well as [table driven testing](https://dave.cheney.net/2019/05/07/prefer-table-driven-tests) 
+- Implement ability to convert between Go in memory objects to JSON representation and vice versa using `json.Marshal() and json.Unmarshal()`
+- Implement ability to accept user input from CLI and read data from a file
+- Load data from json text file and store in an instance of storer
+- Learn how to create REST APIs to expose functionality over the network
+- Learn how to test REST APIs with the net/http/httptest package
+- Learning how to use the Go standard library as well as external third party libraries such as [chi](https://godoc.org/github.com/go-chi/chi), [kingpin](https://godoc.org/gopkg.in/alecthomas/kingpin.v2) and [testify](https://godoc.org/github.com/stretchr/testify)
+
+## Prerequisites
+
+-   Install `go 1.12` according to [official installation instruction](https://golang.org/doc/install)
+-   Clone this project outside your `$GOPATH` to enable [Go Modules](https://github.com/golang/go/wiki/Modules)
+-   Install `golangci-lint` according to [instructions](https://github.com/golangci/golangci-lint#local-installation)
+
+## How to build this project?
```

Comment:
These headings are much bigger than normal
```
## Build

## Run

## Test
```

Reply:
Got it